libmscore/slur.cpp

```diff
@@ -372,6 +372,11 @@ void SlurSegment::layoutSegment(const QPointF& p1, const QPointF& p2)
                         continue;
                   if (pp2.x() < x1)
                         break;
+                  // skip start and end segments on assumption start and end points were placed well already
+                  // this avoids overcorrection on collision with own ledger lines and accidentals
+                  // it also avoids issues where slur appears to be attached to a note in a different voice
+                  if (s == slur()->startSegment() || s == slur()->endSegment())
```

Comment:
I don't believe that performance is the main concern now but still `startSegment()` and `endSegment()` search for a segment by tick by looping over all score's segments. Maybe it would be better to call these functions only once outside this loop over the system's segments.

Reply:
Good point, will update.

Reply:
startSegment()/endSegment() can be replaced by startCR()->segment() and endCR()->segment() or the implementation of startSegment() could be optimized this way.